mypy/checker.py

```diff
@@ -609,6 +611,10 @@ def is_implicit_any(t: Type) -> bool:
 
             # Type check body in a new scope.
             with self.binder.top_frame_context():
+                # To be used for super in visit_super in checkexpr
+                # (We don't care when it's not a method)
+                if item.arguments:
+                    self.binder.push(NameExpr('__SelfType'), item.arguments[0].variable.type)
```

Comment:
It's an ugly hack - I remember commenting about this, but I can't find the comment. It has very little to do with the binder. I don't understand your suggestion; how should `checker` communicate this information with `checkexpr`?